pkg/crc/machine/kubeconfig.go

```diff
@@ -198,13 +198,23 @@ func getTokenForUser(username, password, ip string, ca []byte, clusterConfig *ty
 		},
 	}
 	challengeHandler := challengehandlers.NewBasicChallengeHandler(restConfig.Host, "" /* webconsoleURL */, nil /* in */, nil /* out */, nil /* passwordPrompter */, username, password)
-	token, err := tokenrequest.RequestTokenWithChallengeHandlers(restConfig, challengeHandler)
+	token, err := requestTokenWithChallengeHandlers(restConfig, challengeHandler, ingressHTTPSPort)
 	if err != nil {
 		return "", err
 	}
 	return token, nil
 }
 
+func requestTokenWithChallengeHandlers(clientCfg *restclient.Config, handler *challengehandlers.BasicChallengeHandler, port uint) (string, error) {
+	o, err := tokenrequest.NewRequestTokenOptions(clientCfg, false).WithChallengeHandlers(handler)
+	if err != nil {
+		return "", err
+	}
+
+	o.Issuer = fmt.Sprintf("%s:%d", o.Issuer, port)
```

Comment:
thanks, will change this.

Reply:
This was mistake because `o.Issuer` is URL and not the host string and because of that if take it as ip6 and fails
```
ERRO Cannot update kubeconfig: parse "[https://oauth-openshift.apps-crc.testing]:443": first path segment in URL cannot contain colon 
```